src/Dapr.AspNetCore/TwilioWebHookMiddleware.cs

```diff
@@ -0,0 +1,139 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Net.Http.Headers;
+    using System.Text;
+    using System.Text.Json;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.WebUtilities;
+    internal class TwilioWebHookMiddleware
+    {
+        private const string ContentType = "application/x-www-form-urlencoded";
+        private readonly RequestDelegate next;
+
+        public TwilioWebHookMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public Task InvokeAsync(HttpContext httpContext)
+        {
+            // This middleware unwraps any requests with a cloud events (JSON) content type
+            // and replaces the request body + request content type so that it can be read by a
+            // non-cloud-events-aware piece of code.
+            //
+            // This corresponds to cloud events in the *structured* format:
+            // https://github.com/cloudevents/spec/blob/master/http-transport-binding.md#13-content-modes
+            //
+            // For *binary* format, we don't have to do anything
+            //
+            // We don't support batching.
+            //
+            // The philosophy here is that we don't report an error for things we don't support, because
+            // that would block someone from implementing their own support for it. We only report an error
+            // when something we do support isn't correct.
+            if (!this.MatchesContentType(httpContext, out var charSet))
+            {
+                return this.next(httpContext);
+            }
+
+            return this.ProcessBodyAsync(httpContext, charSet);
+        }
+
+        private async Task ProcessBodyAsync(HttpContext httpContext, string charSet) 

```

Comment:
This code is taking query params and creating a json from it and putting it in the request body, the original request body if any is overwritten. What happens if user wants to keep the body? OR in this case there will not be any body.
This seems very specific to a use case and not general.

The original call can directly go to the api and the api can act on it. It can get query strings and json body.

Reply:
I used Twilio as a specific name as the POSTS from Twilio are FormURLCoded without arrays... if I had confidence that this code would convert all FormURL to a JSON format, then I would rename the middle ware to UseFormURLEncoded... but as  you say, we an talk about it tomorrow